docs/plugins.md

```diff
@@ -14,183 +14,188 @@ Looking for help using Snowpack in your project?
+#### Local plugin
 
-```ts
-import { SnowpackPlugin } from 'snowpack/dist-types/config';
+If you need a local plugin just for your own needs, add it in your `snowpack.config.json`:
 
-export default function mySnowpackPlugin() {
-  const plugin: SnowpackPlugin = { /* … */ };
-  return plugin;
+```json
+{
+  "plugins": [
+    ["../path-to-my-plugin", { optionA: "foo", optionB: "bar" }]
+  ]
 }
 ```
 
-## JavaScript API
+#### npm plugin
 
-### Overview
+To release a plugin to npm, take a look at [snowpack-plugin-starter-template](https://github.com/pikapkg/snowpack-plugin-starter-template) which can get you up-and-running quickly.
 
-```js
-// TODO: More documentation coming soon.
-```
+If rolling your own, however, make sure to mind the following checklist:
 
-### install()
-```js
-// TODO: More documentation coming soon.
-```
+- ✔️ Your `package.json` file has a `main` entry pointing to the final build
+- ✔️ Your code is compiled to run on Node >= 10
+- ✔️ Your package README contains a list of custom options, if your plugin is configurable
 
-### build()
-```js
-// TODO: More documentation coming soon.
-```
+### Tips / Gotchas
 
+- Snowpack’s plugin system is **single-pass** to keep things speedy. That means if you generate a new `.css` file, that file won’t be run back through another CSS plugin.
+- Extensions in Snowpack always have a leading `.` character (e.g. `.js`, `.ts`). This is to match Node’s `path.extname()` behavior, as well as make sure we’re not matching extension substrings (e.g. if we matched `js` at the end of a file, we also don’t want to match `.mjs` files by accident; we want to be explicit there).
+- You may have seen a `transform()` function in previous versions of the plugin API. Though that’ll be supported throughout Snowpack `2.x`, it’s now deprecated and will be removed in the next major release. Instead, please use `build()`.
+- The `input` and `output` file extension arrays are the keys to their simplicity. In other plugin systems you have to parse out the filename to determine whether you want to transform it or not. But with Snowpack, you only have to declare file extensions you’d like to listen for, and Snowpack handles the rest.
+- If `build()` doesn’t return anything, the file isn’t transformed. In this way you can build pure side-effect plugins if needed.
+- If you want to build a plugin that only runs some code on initialization (such as `@snowpack/plugin-dotenv`), put your side-effect code inside your main plugin function to execute when initialized. But be sure to return a `{ name }` so that in case of error, users know what went wrong.
+
+## Bundle Plugin
+
+Snowpack’s bundle plugin API is less standardized and more prone to change. Chances are if you are authoring a bundle plugin, it’s for an existing bundler like Rollup or webpack, and we probably want to add it to our officially-supported bundler plugins!
```

Comment:
"Snowpack’s bundle plugin API is still experimental and may change in a future release. See our official bundler plugins for an example of using the current interface."